### PR TITLE
Decouple dotnet-ef from the framework

### LINKED ISSUE #4837: Make the repo build without the src folder
On the CI we want to run tests without src. Tests will get recompiled but the product will be restored from packages.

This repo fails because: 

```
[Building EntityFramework] build-test
[11:29:04][build-test] Can't find file ..\..\src\Microsoft.EntityFrameworkCore.Commands\tools\OperationHandlers.cs
```

Once you fix this, please update `KOREBUILD_NOSRC_EXCLUDE` on the CI

cc @bricelam

### PR DESCRIPTION
**Pull request by bricelam**
Part of #3925, also resolves #4837

@natemcmaster

### CODE REVIEW

File: src/dotnet-ef/Program.cs [hunk 1]
```diff
@@ -51,7 +50,8 @@ public static int Main([NotNull] string[] args)
                     ex = ex.InnerException;
                 }
 
-                if (!(ex is OperationException))
+                if (!(ex is CommandException
+                    || (ex as OperationException)?.Type == "Microsoft.EntityFrameworkCore.Design.OperationException"))
```
Comment (natemcmaster) [inline src/dotnet-ef/Program.cs]: Should this be `Microsoft.EntityFramework.Commands.OperationException`?

File: src/Microsoft.EntityFrameworkCore.Commands/Design/OperationExecutor.cs [hunk 2]
```diff
@@ -134,10 +136,12 @@ public class AddMigration : OperationBase
                 outputDir,
                 contextType);
 
-            // NOTE: First file will be opened in VS
-            yield return files.MigrationFile;
-            yield return files.MetadataFile;
-            yield return files.SnapshotFile;
+            return new Hashtable
```
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/OperationExecutor.cs]: Change return type of function? This doesn't compile.

Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/OperationExecutor.cs]: Looks like we also need to update tests
```
Microsoft.EntityFrameworkCore.Design.Internal.OperationExecutorTest+SimpleProjectTest.AddMigration_works_cross_domain [FAIL]
      System.InvalidCastException : Unable to cast object of type 'System.Collections.Hashtable' to type 'System.Collections.Generic.IEnumerable`1[System.String]'.
```

File: src/dotnet-ef/Program.cs (hunk 1 above)
Comment (bricelam) [inline src/dotnet-ef/Program.cs]: I think this is correct. I'll rename some of the types in Commands, so it's less confusing. (currently there are two `OperationException` and `OperationExecutor` types)

File: src/Microsoft.EntityFrameworkCore.Commands/Design/OperationExecutor.cs (hunk 2 above)
Comment (bricelam) [inline src/Microsoft.EntityFrameworkCore.Commands/Design/OperationExecutor.cs]: Thanks. :smile: I rushed the review out. I was going to actually bash on it today.

**Comment by bricelam** [comment]
I've bashed on it and things look OK. 83e510f987d9c5abf184e33da85ee6577672821b needs review.

Unfortunately, I get the following errors when using the tool with portable apps. :disappointed: Everything appears to be working with standalone apps.

* Unable to load DLL 'sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
* Could not load file or assembly 'System.Data.SqlClient, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

**Comment by natemcmaster** [comment]
> Unfortunately, I get the following errors when using the tool with portable apps.

Possible related to other "portable with native" errors? cref https://github.com/dotnet/cli/issues/1940

**Comment by natemcmaster** [comment]
Don't forget to update csproj

**Comment by natemcmaster** [comment]
I want to say :shipit:, but we should probably first make sure CLI is aware that tools on portable apps w/ native doesn't work.

**Comment by bricelam** [comment]
Will file a bug, but these issues may no longer be relevant after we resolve #4577.

**Comment by natemcmaster** [comment]
ok. As long as dotnet-ef is working on something, this is improvement, right? :shipit: w/ csproj fixes.

**Comment by bricelam** [comment]
>  this is improvement, right?
lol, yes. We've gone from "not working" to "almost working" :wink:

**Comment by bricelam** [comment]
Merged as 595f2623572bde52619df36a5cf739183b7f7260 & 4df6dbe31ac47513166580c595ad7a62776e3465

**Pull request closed without merging**